packages/config-loader/package.json

```diff
@@ -36,15 +36,15 @@
     "@backstage/types": "^0.1.1",
     "@types/json-schema": "^7.0.6",
     "ajv": "^7.0.3",
+    "cross-fetch": "^3.0.6",
```

Comment:
I'd say using `node-fetch` is preferred for Node.js-only packages, as it provides more accurate and convenient types.

Reply:
And this is present only to be able to mock it in tests, is that it? It's sort of a bit unfortunate to require a package _at all_ for such a core thing :)

Reply:
Hmm yes, this could be an argument to keep it. Do we want to align the versions with the one that comes from `cross-fetch` or is that unimportant?

Reply:
Here's an example of what you need to to otherwise when using `cross-fetch`:

https://github.com/backstage/backstage/blob/2ab7e0f688764b24351ca82fe10981bcfc5dbe35/packages/backend-common/src/reading/BitbucketUrlReader.ts#L105

If `backend-common` used `node-fetch` (which I think it could switch to), that would've been just `response.buffer()`

Then again ArrayBuffers are becoming more of a thing in node, so differences will disappear over time, but thinking is still that `node-fetch` will have some QoL improvements over using `cross-fetch` in node

Reply:
Then again I don't know if there's perhaps a reason to stick to `cross-fetch` because of `msw`, is that the case @benjdlambert?

Reply:
... but I can't find one, `Buffer` extends `UInt8Array` so using it as a response with `msw` works just fine :grin:

Reply:
Hmm I can't really remember why we started the unification effort. I think it was because of the mismatch in implementations according to the spec, like the `buffer` thing being available in `node` and being confusing when writing plugins which one to use. It would be good to just have one, especially when we're going to have isomorphic clients and isomorphic packages. I don't have a strong opinion either way to be honest, if it works, and we don't want to have to wrap the `buffer` response all the time then maybe lets use `node-fetch` in places that it feels the most sane, but push for `cross-fetch` as default still?

Reply:
I closed the PR because I accidentally did it from the `master` of our fork... I only noticed that the package uses a different version of `node-fetch` as we use in other places and it has a non-`^` constraint; that looked a bit odd. But I wasn't aware that `node-fetch` actually has a superior interface than the original `fetch` interface. So I don't want to force anyone using one or the other :-)